modules/registrars/openprovider/src/Configuration.php

```diff
@@ -76,16 +76,22 @@ public static function getCssModuleUrl($cssModuleName)
 
     public static function getServerUrl()
     {
-        $systemUrl = localAPI(WHMCSApiActionType::GetConfigurationValue, ['setting' => 'SystemURL'])['value'];
+        if (method_exists('\\WHMCS\\Config\\Setting', 'getValue')) {
+            $systemUrl = rtrim(\WHMCS\Config\Setting::getValue('SystemURL'), '/') . "/";
+        }
+
+        if (!isset($systemUrl)) {
+            $systemUrl = localAPI(WHMCSApiActionType::GetConfigurationValue, ['setting' => 'SystemURL'])['value'];
```

Comment:
I couldn't understand here.
if you set the default value for systemUrl, then no need check here, right?

Reply:
Here we removed setting systemUrl using localAPI since it is little bit time expensive.
Now first we try to set it using a system variable value if it is available. But if it is not, then only we use LocalAPI method to set the value.

Reply:
oh okay.
may be you can add a comment if you want, may be useful in readability